bloom-housing/bloom
# feat: public lottery view, rank data

This PR addresses #4224


## Description

Pulls some of the backend data into the public lottery view screen. This just tackles showing the raw rank and preferences ranks - getting the total number of applicants and number of applicants per preference I'll handle in a separate PR.

## How Can This Be Tested/Reviewed?

You need a bit of data to test this out. I submitted two applications while signed in to a listing with a lottery and preferences. After running the lottery, when you view the lottery results, the correct ranks for raw rank and preference ranks if the application claimed it should show.

Additionally, we will also see the number of available units.

Example:

<img width="266" alt="Screenshot 2024-08-26 at 3 40 11 PM" src="https://github.com/user-attachments/assets/d51e262b-c010-4a8c-b411-c59dd33e6fe5">

The new additions in this PR are `1 unit`, the raw rank, and the preference ranks. The above screenshot is from using a seeded listing locally which had seeded applications which was helpful for testing.

## Author Checklist:

- [ ] Added QA notes to the issue with applicable URLs - will wait to move to QA for the full AC

## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates
